modules/extensions-mule-support/src/main/java/org/mule/runtime/module/extension/mule/internal/loader/parser/utils/AggregatedErrorsCharacteristic.java

```diff
@@ -0,0 +1,296 @@
+                                                    List<ComponentAst> hierarchy) {
+      Optional<OperationModel> operationModel = operationAst.getModel(OperationModel.class);
+      operationModel.ifPresent(model -> model.getErrorModels().stream().map(MuleSdkErrorModelParser::new)
+          .forEach(errorModelParser -> addIfNotSuppressed(errorModelParser, models, hierarchy)));
+    }
+
+    private void handleRaiseError(ComponentAst raiseErrorAst, List<ErrorModelParser> errorModels, List<ComponentAst> hierarchy) {
+      final ComponentParameterAst typeParameter = raiseErrorAst.getParameter(DEFAULT_GROUP_NAME, ERROR_TYPE_PARAM);
+      if (null == typeParameter) {
+        return;
+      }
+
+      Optional<String> errorId = typeParameter.getValue().<String>getValue();
+      if (!errorId.isPresent()) {
+        return;
+      }
+
+      // TODO: Use the extension parser's namespace.
+      addIfNotSuppressed(new MuleSdkErrorModelParser(APP_LOCAL_EXTENSION_NAMESPACE, errorId.get(), null), errorModels, hierarchy);
+    }
+
+    private boolean isRaiseError(ComponentAst operationAst) {
+      return operationAst.getIdentifier().equals(RAISE_ERROR_IDENTIFIER);
+    }
+
+    private void addIfNotSuppressed(ErrorModelParser errorModelParser, List<ErrorModelParser> models,
+                                    List<ComponentAst> hierarchy) {
+      List<ErrorModelParserMatcher> suppressedErrors = getSuppressedErrors(hierarchy);
+      if (!isErrorSuppressed(errorModelParser, suppressedErrors)) {
+        models.add(errorModelParser);
+      }
+    }
+
+    private List<ErrorModelParserMatcher> getSuppressedErrors(List<ComponentAst> hierarchy) {
+      if (hierarchy.isEmpty()) {
+        return emptyList();
+      }
+
+      int containerIndex = hierarchy.size() - 1;
+      List<ErrorModelParserMatcher> suppressedErrors = new ArrayList<>();
+      while (containerIndex >= 0) {
+        ComponentAst container = hierarchy.get(containerIndex);
+        if (container.getIdentifier().equals(TRY_IDENTIFIER)) {
+          addSuppressedErrors(container, suppressedErrors);
+        } else if (container.getIdentifier().equals(ERROR_HANDLER_IDENTIFIER)) {
+          --containerIndex;
+        }
+        --containerIndex;
+      }
+      return suppressedErrors;
+    }
+
+    private void addSuppressedErrors(ComponentAst tryScopeAst, List<ErrorModelParserMatcher> suppressedErrors) {
+      Optional<ComponentAst> errorHandler = tryScopeAst.directChildrenStreamByIdentifier(null, ERROR_HANDLER).findFirst();
+      if (!errorHandler.isPresent()) {
+        return;
+      }
+
+      Optional<ComponentAst> onErrorContinue =
+          errorHandler.get().directChildrenStreamByIdentifier(null, ON_ERROR_CONTINUE).findFirst();
+      if (!onErrorContinue.isPresent()) {
+        return;
+      }
+
+      String typeAsString =
+          (String) onErrorContinue.get().getParameter(DEFAULT_GROUP_NAME, ERROR_TYPE_PARAM).getValue().getValue().orElse(null);
```

Comment:
I don't understand how this doesn't blow up when the `ERROR_TYPE_PARAM` is not present like in your [test](https://github.com/mulesoft/mule/pull/11692/files#diff-ee6a64fa8e3c21dfaf9faac6e8e5ca0a76a59679b9d6eb944f900f3c6c8751efR45).

Initially I would think that `getParameter` would return `null` (as explained in the Javadoc). And immediately, the first `getValue` should throw an NPE.

Do you know what's going on?

Reply:
that's tricky but is probably something to be clarified on the Javadoc of `getParameter`, which I wouldn't do as part of this PR.